### PR TITLE
feat(ui-components): Slider shared component

### PR DESCRIPTION
**Pull request by helgifr**
Making the Slider component  a shared ui-component so it can be used by all application types

https://dit-iceland.atlassian.net/browse/TS-364

# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `Slider` component with support for `rangeDates` and `onChangeEnd` callback.
  - Added a `SliderFormField` component for integrating slider inputs with form handling.

- **Enhancements**
  - Added `SLIDER` field type to expand available form components.
  - Updated several templates to use the new `Slider` component from `@island.is/application/ui-components`.

- **Documentation**
  - Added examples and props documentation for `SliderFormField` in the storybook.

### CODE REVIEW

**Review by Toti91 — COMMENTED** [review]
Looks good but same comment as this https://github.com/island-is/island.is/pull/14688#pullrequestreview-2054942429

**Comment by obmagnusson** [comment]
Could you provide a screen shot of the component in action?

**Comment by helgifr** [comment]
Screenshot 2024-05-16 at 13 37 03

File: libs/application/ui-fields/src/lib/SliderFormField/SliderFormField.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      showMinMaxLabels: true,
+      showToolTip: true,
+      trackStyle: { gridTemplateRows: 8 },
+      calculateCellStyle: () => {
+        return {
+          background: 'blue',
+        }
+      },
+    })`)}
+/>
```
[the remark this answers is not in the document]
Comment (helgifr) [reply libs/application/ui-fields/src/lib/SliderFormField/SliderFormField.stories.mdx:55]: Doesn't make sense

**Comment by andes-it** [comment]
This PR's had a 15-day siesta. Lost its spark, did it? It's making an exit in 5 days unless the stale label disappears.

**Review by ylfahfa — APPROVED** [review]
lgtm 😺

**Review by Toti91 — APPROVED** [review]
LGTM

**Pull request merged**